### ISSUE TITLE
Interpretation of empty publisherTC

### ISSUE THREAD

**Issue by philoou** [InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework#223]
How should an empty publisherTC be interpreted? 

By default CMPs sent an empty  **PublisherTC**.
But, according to [Publisher TC](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md#publisher-purposes-transparency-and-consent) for **PubPurposesConsent**, valid values for each purpose are 0 or 1. Empty value is not documented.

Should that be interpreted as no-consent (ie. empty = 0) or should empty values should be ignored? 
Can the documentation be clarified on this use-case?

Thanks.

**Comment by janwinkler** [comment]
A vendor should not use it at all. It is optional and only meant for the publishers own use. If a publisher wants to use it for its own purposes, a non existing publisherTC or a publisherTC with all 0 should be understood as "no consent" (for the publisher)

**Comment by patrickjae** [comment]
@janwinkler interesting, is there any legal backing to this interpretation?

According to one of our major SSP partners, they would consider an empty PublisherTC string as no consent/user objection and thus would interpret the whole consent as invalid. 
They argue that an empty PublisherTC string would mean "no consent" and thus no legal basis at all for the publisher when calling the SSP which would (as in this situation the publisher would still be the data controller) nullify the existing consent for the SSP that may be present in the core string in the subsequent processing, as they would be passing at least the user's IP address without any legal basis.
However, I'm not sure how valid this argument is.

It seems that CMPs either don't bother about the PublisherTC substring when setting the TCObject or their UIs are very unclear on this and so during encoding, an empty (i.e. all denied) PublisherTC string is added.
Also, it is not clear from the documentation if the PublisherTC string is mandatory in case of a service-specific legal basis.
However, the IAB JS SDK automatically adds a PublisherTC string whenever it sees that isServiceSpecific flag in the core string. 
Maybe someone of the contributors can clarify on this?

**Comment by janwinkler** [comment]
an empty or non-existing publihserTC string does not mean that the whole consent string is invalid. the publisherTC is optional and can therefore be ignored by the vendor. also you could argue that an empty publisherTC means "no consent FOR THE PUBLISHER" but does not exclude consent for other vendors (although unlikely that a visitor chooses this logic).
also the vendor should consider that there are veeeeery few publishers actually knowing what is going on (technically and from the iab policy side), and hence i would expect many tc strings not to include any publisherTC part.

regarding the spec: it is clear the the publisherTC is NOT mandatory (=its optional):
"A service-specific TC String must contain a Core TC String and may *optionally* contain a Publisher TC segment, ..."

**Comment by philoou** [comment]
@janwinkler, IAB defines the PublisherTC segment as "a segment of a TC String that publishers may use to establish transparency with and receive consent from users for their own legal bases to process personal data or to share with vendors if they so choose." (Source: [IAB Europe](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md#publisher-purposes-transparency-and-consent)). The IAB has confirmed to FreeWheel that the CoreTC segments should be read by vendors who are controllers, not processors. We appreciate that the PublisherTC segment is optional, however, processor vendors must rely on the publisher’s (the controller’s) legal bases for processing which (where TCF v2 has been implemented) should be located in the PublisherTC segment. Processors do not have their own legal bases to process personal data.

**Comment by janwinkler** [comment]
yes, when I say "vendor" i mean a separate entity that is not a processor on behalf of the publisher

**Comment by patrickjae** [comment]
@janwinkler thanks for the clarification, agree on all your points, must have read over the "optionally"...

The issue we see seems to be based on the behavior of the SDK https://github.com/InteractiveAdvertisingBureau/iabtcf-es which includes the PublisherTC string by default to the encoded string, even if no publisher object is present in the TCObject, thus creating an empty PublisherTC string segments whenever the core string is service specific. Will raise an issue there.

**Comment by janwinkler** [comment]
@patrickjae sharing the IP is covered under special purpose 2, which does not need consent and is not signaled in the tcstring

**Comment by patrickjae** [comment]
@janwinkler there is no option to opt-out of special purpose 2, although a legitimate interest legal basis would need to be established in the first place also for it (and in this case by the publisher)

Coming back to the separation of data controller/data processor, in the case of an SSP or other marketing partner (who both would be processors acting on the behalf of the publisher), the legal basis which needs to be relied upon is that of the data controller. In this case this would be the publisher.
However, if the publisherTC string is empty or doesn't exist (as it is optional), what exactly would then be the legal basis on which a processor could rely upon?

On another note, the publisherTC string allows to encode consent/no consent on 24 purposes, which are supposed to be the ones as laid out in the GVL. However, there are not 24 different purposes defined there. Would you be able to shed some light on this as well?

**Comment by alextcone** [comment]
Preface: this is not legal advice

The policy states: 

> A Publisher must not use Publisher-specific custom Legal Bases Signals to formally or informally agree signalling with any Vendor for Purposes that the Framework covers. 

Any publisher purposes / bases established by custom publisher purposes must not overlap with the standard purposes of the TCF. I understand Freewheel in this case considers itself a vendor acting as a Processor. However, regardless of its consideration it is still subject to ePrivacy which is covered by Purpose 1 of the TCF. Since the publisher cannot create its own custom purpose that covers the same area of Purpose 1 Freewheel can look to whatever is set for Purpose 1 in the string to access a device. It’s highly possible too that the publisher is not using the custom purposes feature of TCF so if you are expecting that to be there at scale you might reset your expectations. Additionally, ad a Processor, you would need to know what the custom publisher purpose(s) was in advance to have your systems configured to only process for that purpose(s). That implies a high level of coordination with each publisher sending this value. Therefore if you are not getting a value for publisherTC from any of the publishers you’ve worked to preconfigure special processing purposes outside of those which the TCF already covers, you should reach out to that pub to ask “what’s going on?”

That said, we should add clarification for CMPs working with publishers choosing to use the custom pub purposes feature.

**Comment by patrickjae** [comment]
@alextcone I was under the impression that the PublisherTC string encodes consent/no consent for both purposes within the framework (which I was referring to in the discussion) _and_ custom publisher purposes.

P.S. not seeking legal advice here, just trying to fully understand the policies :-)

**Comment by asr-enid** [comment]
thats correct - see here - https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md#publisher-purposes-transparency-and-consent

**Comment by alextcone** [comment]
We likely need to rethink how we’ve documented this. Because we’ve made pub segment optional it’s highly likely it’s not being heavily used. Therefore if a vendor considering itself a Processor is hoping to read at least what purpose 1 has for the vendor then they are going to be out of luck in most instances. I’m going to bring this up with the commit group and IAB Europe.

**Comment by asr-enid** [comment]
> Therefore if a vendor considering itself a Processor is hoping to read at least what purpose 1 has for the vendor then they are going to be out of luck in most instances. I’m going to bring this up with the commit group and IAB Europe.

Correct, Vendors that act as processor will want to access this segment to inform themselves on the legal basis in relation to the publisher/advertiser.
